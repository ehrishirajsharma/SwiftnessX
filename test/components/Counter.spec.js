import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Counter from '../../app/components/Counter';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const actions = {
    increment: spy(),
    incrementIfOdd: spy(),
    incrementAsync: spy(),
    decrement: spy()
  };
  const component = shallow(<Counter counter={1} {...actions} />);
  return {
    component,
    actions,
    buttons: component.find('button'),
    p: component.find('.counter')
  };
}

describe('Counter component', () => {
  it('should should display count', () => {
    const { p } = setup();
    expect(p.text()).toMatch(/^1$/);
  });

  it('should first button should call increment', () => {
    const { buttons, actions } = setup();
    buttons.at(0).simulate('click');
    expect(actions.increment.called).toBe(true);
  });

  it('should match exact snapshot', () => {
    const { actions } = setup();
    const counter = (
      <div>
        <Router>
          <Counter counter={1} {...actions} />
        </Router>
      </div>
    );
    const tree = renderer.create(counter).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should second button should call decrement', () => {
    const { buttons, actions } = setup();
    buttons.at(1).simulate('click');
    expect(actions.decrement.called).toBe(true);
  });

  it('should third button should call incrementIfOdd', () => {
    const { buttons, actions } = setup();
    buttons.at(2).simulate('click');
    expect(actions.incrementIfOdd.called).toBe(true);
  });

  it('should fourth button should call incrementAsync', () => {
    const { buttons, actions } = setup();
    buttons.at(3).simulate('click');
    expect(actions.incrementAsync.called).toBe(true);
  });
});
