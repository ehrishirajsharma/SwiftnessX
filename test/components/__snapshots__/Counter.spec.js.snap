// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should match exact snapshot 1`] = `
<div>
  <div>
    <div
      className="backButton"
      data-tid="backButton"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="fa fa-arrow-left fa-3x"
        />
      </a>
    </div>
    <div
      className="counter counter"
      data-tid="counter"
    >
      1
    </div>
    <div
      className="btnGroup"
    >
      <button
        className="btn"
        data-tclass="btn"
        onClick={[Function]}
      >
        <i
          className="fa fa-plus"
        />
      </button>
      <button
        className="btn"
        data-tclass="btn"
        onClick={[Function]}
      >
        <i
          className="fa fa-minus"
        />
      </button>
      <button
        className="btn"
        data-tclass="btn"
        onClick={[Function]}
      >
        odd
      </button>
      <button
        className="btn"
        data-tclass="btn"
        onClick={[Function]}
      >
        async
      </button>
    </div>
  </div>
</div>
`;
